src/spyglass/common/common_optogenetics.py

```diff
@@ -0,0 +1,377 @@
+
+
+@schema
+class VirusInjection(SpyglassMixin, dj.Manual):
+    definition = """
+    # Virus injection information
+    -> Session
+    injection_object_id: varchar(64)  # object id of the injection
+    ---
+    -> Virus
+    name: varchar(255)  # name of the injection
+    description: varchar(255)  # description of the injection
+    hemisphere: enum('left', 'right')  # hemisphere of the injection
+    location: varchar(255)  # location of the injection
+    ap_location: float # anterior-posterior location of the injection (in mm)
+    ml_location: float # medial-lateral location of the injection (in mm)
+    dv_location: float # dorsal-ventral location of the injection (in mm)
+    pitch: float # pitch angle of the injection (in degrees)
+    roll: float # roll angle of the injection (in degrees)
+    yaw: float # yaw angle of the injection (in degrees)
+    volume: float # volume of the injection (in uL)
+    titer: float # titer of the virus (in vG/ml)
+    """
+
+    _nwb_table = Nwbfile
+
+    def insert_from_nwb_object(self, nwb_file_name, virus_injection_object):
+        """
+        Insert a VirusInjection entry from an NWB object.
+
+        Parameters
+        ----------
+        nwb_file_name : str
+            The name of the NWB file containing the virus injection data.
+        virus_injection_object : OptogeneticVirusInjection
+            An instance of the OptogeneticVirusInjection class containing the
+            virus injection data.
+        """
+        key = dict(
+            nwb_file_name=nwb_file_name,
+            virus_name=virus_injection_object.virus.construct_name,
+            injection_object_id=virus_injection_object.object_id,
+            name=virus_injection_object.name,
+            description=virus_injection_object.description,
+            hemisphere=virus_injection_object.hemisphere,
+            location=virus_injection_object.location,
+            ap_location=virus_injection_object.ap_in_mm,
+            ml_location=virus_injection_object.ml_in_mm,
+            dv_location=virus_injection_object.dv_in_mm,
+            pitch=virus_injection_object.pitch_in_deg,
+            roll=virus_injection_object.roll_in_deg,
+            yaw=virus_injection_object.yaw_in_deg,
+            volume=virus_injection_object.volume_in_uL,
+            titer=virus_injection_object.virus.titer_in_vg_per_ml,
+        )
+
+        Virus().insert_from_nwb_object(virus_injection_object.virus)
+        self.insert1(
+            key,
+        )
+
+    def make(self, key):
+        # for use with populate_all_common
+        nwb = (Nwbfile() & key).fetch_nwb()[0]
+        for obj in nwb.objects.values():
+            if isinstance(obj, OptogeneticVirusInjection):
+                self.insert_from_nwb_object(key["nwb_file_name"], obj)
+
+
+@schema
+class OpticalFiberDevice(SpyglassMixin, dj.Manual):
+    definition = """
+    # Optical fiber device information
+    fiber_name: varchar(255)  # name of the device
```

Comment:
That's a large primary key (#630 ) - do you expect this table to ever be fk-ref'd?